### PR TITLE
Remove deprecated string module functions

### PR DESCRIPTION
**Pull request by gedeck**
This PR addresses issue #1222. All uses of the string module were reviewed and replaced with string functions. Most changes were in the sping module, so tests are not picking these up. 

In addition, there were a few type tests for strings that are replace with isinstance.

There is also one additional file in gitignore that is created during a newly added test. `/Code/GraphMol/Depictor/test_data/constrainedCoords.out.sdf`

### CODE REVIEW

**Comment by bp-kelley** [comment]
This slipped by because there do not appear to be any regression tests.  Is this code actively used anymore?

**Comment by gedeck** [comment]
It is. I wonder if there is an alternative. The code is no longer under
development.

**Comment by greglandrum** [comment]
Yeah, it hasn't been maintained for quite a while. I'm not aware of anything else quite as flexible.

I would prefer (and am working towards) having the drawing code available and at least mostly complete in c++. If that can produce svg and png then I think we can deprecate the sping code and anything that uses it. 
That would be one of those "non-backwards compatible changes" that I have discussed.

File: rdkit/sping/util/HTMLPiddler.py [hunk 1]
```diff
@@ -195,7 +197,7 @@ def send_line_break(self):
     self.x = self.indent
     self.atbreak = 0
     if TRACE:
-      raw_input('lb')
+      input('lb')
```
Comment (greglandrum) [inline rdkit/sping/util/HTMLPiddler.py]: I don't think it matters here, but I don't think `input()` in python2 is really the same as `raw_input()` in python2.

**Pull request merged**